src/pages/api/answer.ts

```diff
@@ -1,23 +1,44 @@
 import {NextApiRequest, NextApiResponse} from 'next'
 import {convertkitAxios} from 'utils/axios-convertkit-api'
 import fetchConvertkitSubscriberFromServerCookie from 'utils/fetch-convertkit-subscriber'
+import isEmpty from 'lodash/isEmpty'
+import find from 'lodash/find'
 
 const answer = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === 'POST') {
     try {
-      console.log(req.body)
-      const {tagId} = req.body
+      const {tagId, survey} = req.body
       const cookieHeader = req.headers.cookie as string
-      const [
-        subscriber,
-        ckCookie,
-      ] = await fetchConvertkitSubscriberFromServerCookie(cookieHeader)
+      const siteNameShort = process.env.NEXT_PUBLIC_SITE_NAME_SHORT
+      const [subscriber, ckCookie] =
+        await fetchConvertkitSubscriberFromServerCookie(cookieHeader)
+      const questionId = siteNameShort
+        ? `${siteNameShort}-survey-${survey.id}`
+        : `survey-${survey.id}`
 
+      // Subscribe user to tag
       await convertkitAxios.post(`/tags/${tagId}/subscribe`, {
         api_key: process.env.NEXT_PUBLIC_CONVERTKIT_TOKEN,
         email: subscriber.email_address,
       })
 
+      // Create question field if it doesn't exist
+      const fieldExists = !isEmpty(
+        find(Object.keys(subscriber.fields), (field) => field === questionId),
+      )
+      if (!fieldExists) {
+        await convertkitAxios.post(`/custom_fields`, {
+          api_secret: process.env.CONVERTKIT_API_SECRET,
+          label: questionId,
+        })
+      }
```

Comment:
this section feels like it might make sense as a utility function that takes the subscriber's fields as an argument?